src/twisted/internet/abstract.py

```diff
@@ -269,27 +273,31 @@ def doWrite(self):
         if self.offset == len(self.dataBuffer) and not self._tempDataLen:
             self.dataBuffer = b""
             self.offset = 0
-            # stop writing.
```

Comment:
This is the optimization I worry about a little. What if `resumeProducing()` doesn't write? (I tried conditionally calling `stopWriting()` right after `resumeProducing()` but that makes some tests freeze...)

Reply:
I don't understand the issue / question here.

-------

I think that we should not expect that `resumeProducing` will write new data right away.

It might produce data somewhere in the future, or we might have the connection closed without more data being produced.

Reply:
Given that assumption, this change is wrong.